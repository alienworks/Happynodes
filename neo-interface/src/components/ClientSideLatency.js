import axios from 'axios'
import React, { Component } from 'react'
import Ping from 'ping.js';

class ClientSideLatency extends Component {
    constructor(props) {
        super(props);
        this.state = {
            latency: 0
        }

        this.getLatency = this.getLatency.bind(this);
    }

    getLatency() {
        var p = new Ping();
        const url = this.props.url;

        var t0 = performance.now();

        p.ping(url, (err, data) => {
            var t1 = performance.now();
            this.setState({ latency: (t1 - t0) });
        });
    }

    componentDidMount() {
        this.getLatency()
    }

    render() {
        return (
            <div className="infoblock">
                <h4>{(this.state.latency.toFixed(0)).toLocaleString()}</h4>
                <h5>Live Client Side Latency (ms)</h5>
            </div>
        );
    }
}

export default ClientSideLatency;