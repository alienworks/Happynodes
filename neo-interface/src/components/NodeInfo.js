import React, { Component } from 'react'
import { connect } from 'react-refetch'
import emojiFlags from 'emoji-flags';
import ClientSideLatency from './ClientSideLatency';
import config from './config'

class NodeInfo extends Component {
    render() {
        const { nodeinfo } = this.props;
        
        if (nodeinfo.pending) {
            return (

                <div className="jumbotron">
                    <h2>Loading...</h2>
                </div>
            )
        } else if (nodeinfo.rejected) {
            return (

                <div>
                    error
                </div>
            )
        } else if (nodeinfo.fulfilled) {
            console.log("nodeinfo", nodeinfo)

            const data = nodeinfo.value;

            const boxStyle = {
                'backgroundColor': 'hsl(' + data.health_score + ',100%,50%)',
                'float': 'left',
                padding: '2rem',
                margin: '4px 2rem 2rem 0px',
            }
            const gap = (data.max_blockheight > data.blockheight) ? "(" + (data.blockheight - data.max_blockheight) + ")" : "";


            return (
                <div className="jumbotron nodes" style={{ display: 'inline-block', width: '100%' }}>
                    <div className="node-box" style={boxStyle}><p>{Number(data.health_score.toFixed(0))}</p></div>
                    <h2 style={{ float: 'left', 'overflowWrap': 'break-word', 'width': '80%' }}>{data.address}</h2>

                    <hr style={{
                        'clear': 'left',
                        'borderColor': '#78cadd', 'borderWidth': '3px'
                    }} />

                    <h3>Node Status Information</h3>
                    <hr />

                    <div className="infoblock">
                        <h4>RPC</h4>
                        <h5>Type</h5>
                    </div>
                    <div className="infoblock">
                        <h4>{Number(data.blockheight).toLocaleString()} {gap}</h4>
                        <h5>Current Block</h5>
                    </div>
                    <div className="infoblock">
                        <h4>{(Number(data.connection_counts)).toLocaleString()}</h4>
                        <h5>Peer Nodes</h5>
                    </div>

                    <div className="infoblock">
                        <h4>{Number((Number(data.latency)).toFixed(0)).toLocaleString()}</h4>
                        <h5>Server Side Latency (ms)</h5>
                    </div>

                    <ClientSideLatency url={data.address} />

                    <div className="infoblock">
                        <h4>{data.version}</h4>
                        <h5>Version</h5>
                    </div>

                    <div className="infoblock">
                        <h4>{data.min_ts}</h4>
                        <h5>First Recorded</h5>
                    </div>

                    <div className="infoblock">
                        <h4>{data.wallet_status}</h4>
                        <h5>Wallet Status</h5>
                    </div>

                    {data.p2p_tcp_status ? (
                        <div className="infoblock">
                            <h4>online</h4>
                            <h5>P2P TCP Status</h5>
                        </div>
                    ) : (
                            <div className="infoblock">
                                <h4>offline</h4>
                                <h5>P2P TCP Status</h5>
                            </div>
                        )}

                    {data.p2p_ws_status ? (
                        <div className="infoblock">
                            <h4>online</h4>
                            <h5>P2P WS Status</h5>
                        </div>
                    ) : (
                            <div className="infoblock">
                                <h4>offline</h4>
                                <h5>P2P WS Status</h5>
                            </div>
                        )}


                    {data.rcp_http_status ? (
                        <div className="infoblock">
                            <h4>online</h4>
                            <h5>RCP HTTP Status</h5>
                        </div>
                    ) : (
                            <div className="infoblock">
                                <h4>offline</h4>
                                <h5>RCP HTTP Status</h5>
                            </div>
                        )}

                    {data.rcp_https_status ? (
                        <div className="infoblock">
                            <h4>online</h4>
                            <h5>RCP HTTPs Status</h5>
                        </div>
                    ) : (
                            <div className="infoblock">
                                <h4>offline</h4>
                                <h5>RCP HTTPs Status</h5>
                            </div>
                        )}

                    <div className="infoblock" style={{ clear: 'left' }}>
                        <h4>{data.mempool_size}</h4>
                        <h5>Unconfirmed Transactions</h5>
                    </div>
                    <div className="infoblock">
                        <h4>{data.stability}%</h4>
                        <h5>Stability (Last 100 pings)</h5>
                    </div>
                    <div className="infoblock">
                        <h4>{data.validated_peers_counts}</h4>
                        <h5>Validated Peers</h5>
                    </div>

                    <div className="infoblock">
                        <h4>{emojiFlags.countryCode(data.locale).name} ({emojiFlags.countryCode(data.locale).emoji})</h4>
                        <h5>Country</h5>
                    </div>

                    <div className="infoblock">
                        <h4>{data.last_update_time}%</h4>
                        <h5>Last Update Time</h5>
                    </div>

                    <h3 style={{ paddingTop: '4rem', clear: 'left' }}>Happynodes Score</h3>
                    <h5>Score is calculated by taking the average of four metrics</h5>
                    <hr />

                    <div className="infoblock">
                        <h4>{Number(data.health_score.toFixed(0))}%</h4>
                        <h5>Health Score</h5>
                    </div>

                    <div className="infoblock">
                        <h4>:</h4>

                    </div>




                    <div className="infoblock">
                        <h4>{Number(data.blockheight_score).toFixed(0)}%</h4>
                        <h5>Blockheight Lag</h5>
                    </div>
                    <div className="infoblock">
                        <h4>{Number(data.validated_peers_counts_score).toFixed(0)}%</h4>
                        <h5>Validated Peers</h5>
                    </div>
                    <div className="infoblock">
                        <h4>{Number(data.stability).toFixed(0)}%</h4>
                        <h5>Stability (100 pings)</h5>
                    </div>
                    <div className="infoblock">
                        <h4>{Number(data.normalised_latency_score.toFixed(0))}%</h4>
                        <h5>Latency</h5>
                    </div>


                </div>
            )
        }
    }
}

export default connect((props) => ({
    nodeinfo: { url: config.api_url.concat(`/nodes/${props.node_id}`) },
}))(NodeInfo)