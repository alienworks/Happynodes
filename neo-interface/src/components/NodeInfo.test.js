import React from 'react';
import ReactDOM from 'react-dom';
import NodeInfo from './NodeInfo';
import ClientSideLatency from './ClientSideLatency';
import renderer from 'react-test-renderer';

Date.now = jest.fn(() => 1487076708000)

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NodeInfo />, div);
    ReactDOM.unmountComponentAtNode(div);
});


describe("test against valid snapshot", () => {
    it("redner as expected", () => {
        const tree = renderer.create(<NodeInfo nodeinfo={{ fulfilled:true ,value: { "id": 42, "hostname": "seed3.aphelion-neo.com", "protocol": "http", "port": 10332, "p2p_tcp_status": true, "p2p_ws_status": true, "address": "http://seed3.aphelion-neo.com:10332", "validated_peers_counts": 18, "stability": 100, "blockheight_score": 98, "normalised_latency_score": 87.9701137542725, "validated_peers_counts_score": 34.61538461538461, "health_score": 80.1463745924143, "latency": 240.597724914551, "rcp_https_status": true, "rcp_http_status": true, "mempool_size": 74, "connection_counts": 244, "online": true, "blockheight": 2622828, "lat": 50.1153, "long": 8.6823, "locale": "kr", "version": "/NEO:2.7.4/", "max_blockheight": 2622829 } }} />).toJSON();

        expect(tree).toMatchSnapshot();

    });
});
