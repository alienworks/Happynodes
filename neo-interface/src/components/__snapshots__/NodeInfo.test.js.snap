// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test against valid snapshot redner as expected 1`] = `
<div
  className="jumbotron nodes"
  style={
    Object {
      "display": "inline-block",
      "width": "100%",
    }
  }
>
  <div
    className="node-box"
    style={
      Object {
        "backgroundColor": "hsl(80.1463745924143,100%,50%)",
        "float": "left",
        "margin": "4px 2rem 2rem 0px",
        "padding": "2rem",
      }
    }
  >
    <p>
      80
    </p>
  </div>
  <h2
    style={
      Object {
        "float": "left",
        "overflowWrap": "break-word",
        "width": "80%",
      }
    }
  >
    http://seed3.aphelion-neo.com:10332
  </h2>
  <hr
    style={
      Object {
        "borderColor": "#78cadd",
        "borderWidth": "3px",
        "clear": "left",
      }
    }
  />
  <h3>
    Node Status Information
  </h3>
  <hr />
  <div
    className="infoblock"
  >
    <h4>
      RPC
    </h4>
    <h5>
      Type
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      2,622,828
       
      (-1)
    </h4>
    <h5>
      Current Block
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      244
    </h4>
    <h5>
      Peer Nodes
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      241
    </h4>
    <h5>
      Server Side Latency (ms)
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      0
    </h4>
    <h5>
      Live Client Side Latency (ms)
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      /NEO:2.7.4/
    </h4>
    <h5>
      Version
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      online
    </h4>
    <h5>
      P2P TCP Status
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      online
    </h4>
    <h5>
      P2P WS Status
    </h5>
  </div>
  <div
    className="infoblock"
    style={
      Object {
        "clear": "left",
      }
    }
  >
    <h4>
      74
    </h4>
    <h5>
      Unconfirmed Transactions
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      100
      %
    </h4>
    <h5>
      Stability (Last 100 pings)
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      18
    </h4>
    <h5>
      Validated Peers
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      South Korea
       (
      ðŸ‡°ðŸ‡·
      )
    </h4>
    <h5>
      Country
    </h5>
  </div>
  <h3
    style={
      Object {
        "clear": "left",
        "paddingTop": "4rem",
      }
    }
  >
    Happynodes Score
  </h3>
  <h5>
    Score is calculated by taking the average of four metrics
  </h5>
  <hr />
  <div
    className="infoblock"
  >
    <h4>
      80
      %
    </h4>
    <h5>
      Health Score
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      :
    </h4>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      98
      %
    </h4>
    <h5>
      Blockheight Lag
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      35
      %
    </h4>
    <h5>
      Validated Peers
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      100
      %
    </h4>
    <h5>
      Stability (100 pings)
    </h5>
  </div>
  <div
    className="infoblock"
  >
    <h4>
      88
      %
    </h4>
    <h5>
      Latency
    </h5>
  </div>
</div>
`;
